v0/destinations/singular/transform.js

```diff
@@ -0,0 +1,215 @@
+  const eventType = message.event;
+  const response = defaultRequestConfig();
+  const eventAttributes = {};
+  let payload, endPoint;
+  const platform = getValueFromMessage(message, "context.os.name");
+  if (
+    Config.sessionEventList.includes(eventType) ||
+    sessionEvents.includes(eventType)
+  ) {
+    if (platform.toLowerCase() === "android") {
+      payload = constructPayload(
+        message,
+        MAPPING_CONFIG[CONFIG_CATEGORIES.SESSION_ANDROID.name]
+      );
+      if (!payload) {
+        // fail-safety for developer error
+        throw new CustomError("Failed to Create Android Session Payload", 400);
+      }
+      extractCustomFields(
+        message,
+        eventAttributes,
+        ["properties"],
+        SINGULAR_SESSION_ANDROID_EXCLUSION
+      );
+    } else if (platform.toLowerCase() === "ios") {
+      payload = constructPayload(
+        message,
+        MAPPING_CONFIG[CONFIG_CATEGORIES.SESSION_IOS.name]
+      );
+      if (!payload) {
+        // fail-safety for developer error
+        throw new CustomError("Failed to Create iOS Session Payload", 400);
+      }
+      extractCustomFields(
+        message,
+        eventAttributes,
+        ["properties"],
+        SINGULAR_SESSION_IOS_EXCLUSION
+      );
+
+      /*
+        if att_authorization_status is true then dnt will be false,
+        else by default dnt value is true
+      */
+      payload.dnt = !payload.att_authorization_status;
+    } else {
+      throw new CustomError("[Singular] :: Invalid Platform", 400);
+    }
+
+    // Singular maps Connection Type to either wifi or carrier
+    if (message.context?.network?.wifi === "wifi") {
+      payload.c = "wifi";
+    } else {
+      payload.c = "carrier";
+    }
+
+    payload = removeUndefinedAndNullAndEmptyValues(payload);
+    endPoint = `${BASE_URL}/launch`;
+  } else {
+    // event payload
+    if (platform.toLowerCase() === "android") {
+      payload = constructPayload(
+        message,
+        MAPPING_CONFIG[CONFIG_CATEGORIES.EVENT_ANDROID.name]
+      );
+      if (!payload) {
+        // fail-safety for developer error
+        throw new CustomError("Failed to Create Android Event Payload", 400);
+      }
+      extractCustomFields(
+        message,
+        eventAttributes,
+        ["properties"],
+        SINGULAR_EVENT_ANDROID_EXCLUSION
+      );
+    } else if (platform.toLowerCase() === "ios") {
+      payload = constructPayload(
+        message,
+        MAPPING_CONFIG[CONFIG_CATEGORIES.EVENT_IOS.name]
+      );
+      if (!payload) {
+        // fail-safety for developer error
+        throw new CustomError("Failed to Create iOS Event Payload", 400);
+      }
+      extractCustomFields(
+        message,
+        eventAttributes,
+        ["properties"],
+        SINGULAR_EVENT_IOS_EXCLUSION
+      );
+    } else {
+      throw new CustomError("[Singular] :: Invalid Platform", 400);
+    }
+    const productList = message.properties?.products;
```

Comment:
No need to declare a new variable here. Use object destructuring.
```suggestion
    const { products } = message.properties;
```

Reply:
Done